lib/spack/spack/test/installer.py

```diff
@@ -73,53 +72,21 @@ def _true(*args, **kwargs):
     return True
 
 
-def create_build_task(pkg, install_args={}):
-    """
-    Create a built task for the given (concretized) package
-
-    Args:
-        pkg (spack.package_base.PackageBase): concretized package associated with
-                                              the task
-        install_args (dict): dictionary of kwargs (or install args)
-
-    Return:
-        (BuildTask) A basic package build task
-    """
-    request = inst.BuildRequest(pkg, install_args)
-    return inst.BuildTask(pkg, request, False, 0, 0, inst.STATUS_ADDED, [])
-
-
-def create_installer(installer_args):
-    """
-    Create an installer using the concretized spec for each arg
-
-    Args:
-        installer_args (list): the list of (spec name, kwargs) tuples
+def create_build_task(
+    pkg: spack.package_base.PackageBase, install_args: Optional[dict] = None
+) -> inst.BuildTask:
+    request = inst.BuildRequest(pkg, {} if install_args is None else install_args)
+    return inst.BuildTask(pkg, request, False, 0, 0, inst.STATUS_ADDED, set())
 
-    Return:
-        spack.installer.PackageInstaller: the associated package installer
-    """
-    const_arg = [(spec.package, kwargs) for spec, kwargs in installer_args]
-    return inst.PackageInstaller(const_arg)
 
-
-def installer_args(spec_names, kwargs={}):
-    """Return a the installer argument with each spec paired with kwargs
-
-    Args:
-        spec_names (list): list of spec names
-        kwargs (dict or None): install arguments to apply to all of the specs
-
-    Returns:
-        list: list of (spec, kwargs), the installer constructor argument
-    """
-    arg = []
-    for name in spec_names:
-        spec = spack.spec.Spec(name)
-        spec.concretize()
-        assert spec.concrete
-        arg.append((spec, kwargs))
-    return arg
+def create_installer(
+    specs: Union[List[str], List[spack.spec.Spec]], install_args: Optional[dict] = None
+) -> inst.PackageInstaller:
+    """Create an installer instance for a list of specs or package names that will be
+    concretized."""
+    _specs = [spack.spec.Spec(s).concretized() if isinstance(s, str) else s for s in specs]
```

Comment:
We can look in a following PR if we could get some speed-up in tests by using `default_mock_concretization` here.